### ISSUE TITLE
Release a built package

### ISSUE THREAD

**Issue by cyberb** [standardnotes/web#281]
Could you please create a package that is ready for deployment.
I assume it should include the result of npm run build for front end and needed code for backend.

**Comment by mobitar** [comment]
We do deploy each version as an npm package. That's how we get it inside the desktop app. See package.json for name and version info: https://github.com/standardnotes/web/blob/master/package.json#L2

**Comment by cyberb** [comment]
Currently if I try to run npm run build it tries to build phantomjs and also need ruby:
    Warning: You need to have Ruby and Haml installed and in your PATH

If I can skip building it by downloading standard-notes-web npm package, is it enough to run it or I still need ruby server part of it?

Also Is this ruby server part the same as standard files server (which I have running using go implementation) or nor not?

**Comment by mobitar** [comment]
If you install from npm you don't need to build it. The built files will be in the dist directory. The ruby server aspect in this repository is just the wrapping server to deploy and manage releases with Capistrano and Passenger. For the npm package you don't need to use this server.

**Comment by cyberb** [comment]
Can I run it with nginx?

**Comment by mobitar** [comment]
Yup..https://docs.standardnotes.org/self-hosting/self-hosting-with-ec2-and-nginx (for sync server, but may be useful as inspiration).

**Comment by cyberb** [comment]
I have npm package downloaded with the following contents:

root@server-new:~/notes/node_modules/standard-notes-web/dist# ls

assets  javascripts  stylesheets

What should my nginx root be?

**Comment by mobitar** [comment]
Come to think about it, the npm package doesn't include an index.html page. What you'd want to do is create a static web project that installs this package, and add an index.html file in the root and import javascripts/compiled.js and stylesheets/app.css.

Our index.html:
Web: https://github.com/standardnotes/web/blob/master/app/views/application/app.html.erb#L37
Desktop: https://github.com/standardnotes/desktop/blob/master/app/index.html#L67

**Comment by cyberb** [comment]
Ok, everything seem to work.

It would be still great to have a versioned archive somewhere like github releases containing the following:
1. Web files so you do not have to use npm/node to get it.
2. Index page for browser to use, ideally pointing at api as a default same host sf implementation.
3. Default extensions so you do not have to use git.

For reference: https://github.com/syncloud/notes/blob/master/build.sh

Otherwise feel free to close it.

**Comment by mobitar** [comment]
Your script looks good! 

Do you mean a version not using Rails? Because that's what this repo would be. It has the web files, it has the index page, and it includes the default extensions as submodules..

**Comment by cyberb** [comment]
Yes if that makes sense.

**Comment by mobitar** [comment]
Static packages are now available as of standard-notes-web@3.0.10-beta2. Simply open the index.html file in the root of the package, which will access files from the dist directory.

Note that for extensions to work properly, like the Extensions panel itself, the index.html page must be served from an http url, and thus a server, and not a file:// url. **Extensions will not work in browser if the index.html is opened as a file://.**

**Comment by baptiste-grob** [comment]
You're right, we changed the way we use the web dependency. We'll have to look into auto-publishing to npm. 👍
